### PR TITLE
*: move mocktikv from tikv pkg

### PR DESCRIPTION
**Pull request by siddontang**
Now tikv package used mocktikv which introduce parser package and let outside project can't use TiDB in the vendor directly.

So here we decide to move the mocktikv from tikv package to the mockstore package. 

This is the first step and not perfect because tikv test also needs to use mocktikv but mockstore also needs some tikv functions to create the storage. So to break the cycle dependence, here we will use some duplicated code.

Should simply them later after refactoring the whole tikv test. 

PTAL @shenli @coocood @wentaoxu

### CODE REVIEW

File: store/mockstore/test_util.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
+// Copyright 2017 PingCAP, Inc.
```
Comment (shenli) [inline store/mockstore/test_util.go]: 2017 -> 2018

**Comment by coocood** [comment]
LGTM

**Pull request merged**